File: AndroidStudioProjects/HealLotsMobile/app/src/main/java/com/heallots/mobile/api/RequestInterceptor.java
package com.heallots.mobile.api;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;
import java.io.IOException;

public class RequestInterceptor implements Interceptor {
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request originalRequest = chain.request();

        // Add logging if needed
        Request request = originalRequest.newBuilder()
                .header("Content-Type", "application/json")
                .build();

        return chain.proceed(request);
    }
}

File: AndroidStudioProjects/HealLotsMobile/app/src/main/java/com/heallots/mobile/api/RequestInterceptor.kt
package com.heallots.mobile.api

import java.io.IOException
import okhttp3.Interceptor
import okhttp3.Response

class RequestInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
            .newBuilder()
            .header("Content-Type", "application/json")
            .build()

        return chain.proceed(request)
    }
}
